src/geometry.nim

```diff
@@ -93,7 +93,7 @@ type
 
 proc newPoint2D*(u, v: float32): Point2D {.inline.} = Point2D([u, v]) 
 proc newPoint3D*(x, y, z: float32): Point3D {.inline.} = Point3D([x, y, z])
-proc newNormal*(x, y, z: float32): Normal {.inline.} = Normal([x, y, z])
+proc newNormal*(x, y, z: float32): Normal {.inline.} = Normal([x, y, z].normalize)
```

Comment:
I like it, i feel like we should pay much more attention to normal length, because I'm not that sure that all of them are unitary in our code